tests/cpp/test_multidevice_tutorial.cpp

```diff
@@ -0,0 +1,367 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <multidevice/communicator.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/multidevice.h>
+
+namespace nvfuser {
+
+class MultiDeviceTutorial : public MultiDeviceTest {
+ protected:
+  static void SetUpTestSuite() {
+    verbose_ = getNvFuserEnv("TUTORIAL_VERBOSE");
+  }
+
+ protected:
+  static bool verbose_;
+};
+
+bool MultiDeviceTutorial::verbose_ = false;
+
+// The first object we need to introduce is the class `Communicator` which is
```

Comment:
Do we also want to mention how to run this tutorial tests?

Reply:
I added the following comment at the top of the first unit test:
```
// To run those tests, allocate a node with at least 2 GPUs and run:
//
// mpirun -np 2 -x TUTORIAL_VERBOSE=1 tutorial_multidevice
```